fili-core/src/main/java/com/yahoo/bard/webservice/web/OutstandingRateLimitedRateLimitRequestToken.java

```diff
@@ -0,0 +1,145 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.Principal;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * RateLimitRequestToken for successfully bound request.
+ */
+public class OutstandingRateLimitedRateLimitRequestToken extends RateLimitRequestToken {
+    private final String userName;
+    private final AtomicInteger userCount;
+    private final AtomicInteger globalCount;
+    private boolean isBound;
+
+    private static int DISABLED_RATE = -1;
+    private static final Logger LOG = LoggerFactory.getLogger(OutstandingRateLimitedRateLimitRequestToken.class);
+
+    /**
+     * Bind outstanding request to token, or fail and set to unbound.
+     *
+     * @param user  The user that issued the request.
+     * @param requestLimit  The limit on the amount of requests that user can issue.
+     * @param requestLimitGlobal  The limit on the amount of requests any user can issue.
+     * @param userCount  The amount of in-flight requests the user hass issued.
+     * @param globalCount  The total amount of in-flight requests.
+     * @param requestMeter  A meter counting the total amount of accepted requests made.
+     * @param rejectMeter  A meter counting the total number of rejected requests made.
+     * @param requestGlobalCounter  A counter tracking the total number of accepted requests made.
+     */
+    public OutstandingRateLimitedRateLimitRequestToken(Principal user, int requestLimit, int requestLimitGlobal,
+            AtomicInteger userCount, AtomicInteger globalCount, Meter requestMeter, Meter rejectMeter,
+            Counter requestGlobalCounter
```

Comment:
I didn't move that in the refactor because requestLimitGlobal sounded like it should be the responsibility of the RateLimiter, and not each individual request token. Does it make sense to move this code? How can I know when it is "ok" to make a change like this?